include/camp/defines.in.hpp

```diff
@@ -13,25 +13,31 @@ For details about use and distribution, please read LICENSE and NOTICE from
 
 #include <cstddef>
 #include <cstdint>
-#include <string>
 #include <stdexcept>
+#include <string>
+
+// Define CAMP_CONFIG_OVERRIDE to change this on a per-file basis
+#if !defined(CAMP_CONFIG_OVERRIDE)
+#cmakedefine CAMP_ENABLE_OPENMP
+#cmakedefine CAMP_ENABLE_TARGET_OPENMP
+#cmakedefine CAMP_ENABLE_CUDA
+#cmakedefine CAMP_ENABLE_HIP
+#cmakedefine CAMP_ENABLE_SYCL
```

Comment:
Does this enable everything by default?

Reply:
It does not, it tells cmake to overwrite those lines with the values of those variables from the cmake configuration.  There's a similar block of cmakedefine directives in raja's config.hpp.